cmd/postgresql-upgrade-database.rb

```diff
@@ -66,8 +66,8 @@ def postgresql_upgrade_database
       # Following instructions from:
       # https://www.postgresql.org/docs/10/static/pgupgrade.html
       ohai "Upgrading #{name} data from #{pg_version_data} to #{pg_version_installed}..."
-
-      if /#{name}\s+started/.match?(Utils.popen_read("brew", "services", "list"))
+      sys_services = JSON.parse(Utils.popen_read("brew", "services", "info", "--all", "--json"))
+      if sys_services.any? { |svc| svc['name'] == name }
```

Comment:
```suggestion
      services_json_output = Utils.popen_read("brew", "services", "info", "--all", "--json")
      services_json = JSON.parse(services_json_output, object_class: OpenStruct)
      running_service_names = services_json.select(&:running).map(&:name)
      if running_service_names.include?(name)
```

Reply:
Splitting the call out into separate lines definitely easier to read.

The service shows as stopped and not running if the postgres command is spinning, so that's why it's only checking for existence of the services in the list. Perhaps a deeper problem? In a way what this is asking for is to always run the stops before upgrading, but I suppose this check ensures the existence of the service.